### PR TITLE
Updated error message for KMS keys

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

Additional wrapper to help debugging potential AWS issue where permissions for KMS keys are insufficent.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/blockstorage/awsebs/awsebs.go [hunk 1]
```diff
@@ -456,6 +456,9 @@ func (s *EbsStorage) VolumeCreateFromSnapshot(ctx context.Context, snapshot bloc
 
 	volID, err := createVolume(ctx, s.Ec2Cli, cvi, ktags.GetTags(tags))
 	if err != nil {
+		if isVolNotFoundErr(err) {
+			return nil, errors.Wrap(err, "This may indicate insufficient permissions for KMS keys.")
+		}
```
Comment (onkarbhat) [inline pkg/blockstorage/awsebs/awsebs.go:461]: @bathina2 

I took a look  the isVolNotFoundErr

```

func isVolNotFoundErr(err error) bool {
	return isError(err, volumeNotFoundCode)
}
```

```
func isError(err error, code string) bool {
	awsErr, ok := err.(awserr.Error)
	return ok && awsErr.Code() == code
}
```

This the AWS error interface:

```
type Error interface {
	// Satisfy the generic error interface.
	error

	// Returns the short phrase depicting the classification of the error.
	Code() string

	// Returns the error details message.
	Message() string

	// Returns the original error if one was set.  Nil is returned if not set.
	OrigErr() error
}
```

Can you check if there is anything useful returned using Message or OrigErr ? Can we parse that to get the exact reason for the failure during createVolume ?

Comment (onkarbhat) [reply pkg/blockstorage/awsebs/awsebs.go:461]: comment in awserr/error.go :   

```
// Example:
//
//     output, err := s3manage.Upload(svc, input, opts)
//     if err != nil {
//         if awsErr, ok := err.(awserr.Error); ok {
//             // Get error details
//             log.Println("Error:", awsErr.Code(), awsErr.Message())
//
//             // Prints out full error message, including original error if there was one.
//             log.Println("Error:", awsErr.Error())
//
//             // Get original error
//             if origErr := awsErr.OrigErr(); origErr != nil {
//                 // operate on original error.
//             }
//         } else {
//             fmt.Println(err.Error())
//         }
//     }
//
```

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:461]: `"InvalidVolume.NotFound: The volume 'vol-08d83a8945eed3580' does not exist.\\n\\tstatus code: 400, request id: 6b647f82-bcd3-4957-8061-32070bba10dd\"`

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:461]: @onkarbhat refer to this ticket - https://kasten.atlassian.net/browse/K10-2613

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:461]: Thats the full error. Doesn't look like theres any original error set.

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:461]: `

func (r requestError) Error() string {
--
  |   | extra := fmt.Sprintf("status code: %d, request id: [%s]",
  |   | r.statusCode, r.requestID)
  |   | return SprintError(r.Code(), r.Message(), extra, r.OrigErr())
  |   | }

`

Comment (bathina2) [reply pkg/blockstorage/awsebs/awsebs.go:461]: The error output contains the OrigErr.

Comment (onkarbhat) [reply pkg/blockstorage/awsebs/awsebs.go:461]: Thanks for checking @bathina2 . LGTM

**Review by onkarbhat — APPROVED** [review]

**Pull request merged**